File: PARTAGER/Structures/SizeD.cs
namespace FCGP
{
    internal struct SizeD
    {
        private readonly bool FlagNotEmpty;
        internal static SizeD Add(SizeD Taille1, SizeD Taille2)
        {
            return new SizeD(Taille1.Largeur + Taille2.Largeur, Taille1.Hauteur + Taille2.Hauteur);
        }
        internal static SizeD Scale(SizeD Taille, double Facteur)
        {
            return new SizeD(Taille.Largeur * Facteur, Taille.Hauteur * Facteur);
        }
        internal static readonly SizeD Empty;
        internal SizeD(PointD Point)
        {
            Largeur = Point.X;
            Hauteur = Point.Y;
            FlagNotEmpty = true;
        }
        internal SizeD(Size Size)
        {
            Largeur = Size.Width;
            Hauteur = Size.Height;
            FlagNotEmpty = true;
        }
        internal SizeD(double Largeur, double Hauteur)
        {
            this.Largeur = Largeur;
            this.Hauteur = Hauteur;
            FlagNotEmpty = true;
        }
        internal bool IsEmpty
        {
            get
            {
                return !FlagNotEmpty;
            }
        }
        /// <summary> renvoie la largeur du rectangle </summary>
        internal double Largeur { get; private set; }
        /// <summary> renvoie la hauteur du rectangle </summary>
        internal double Hauteur { get; private set; }
        /// <summary> renvoie une size à partir du sizeD avec un arrondi Sup</summary>
        internal Size ToSize
        {
            get
            {
                return new Size((int)Math.Round(Largeur), (int)Math.Round(Hauteur));
            }
        }
        internal Point ToPoint
        {
            get
            {
                return new Point(ToSize);
            }
        }
        /// <summary> renvoie sizeF à partir du sizeD </summary>
        internal SizeF ToSizeF
        {
            get
            {
                return new SizeF((float)Largeur, (float)Hauteur);
            }
        }
        internal PointF ToPoinF
        {
            get
            {
                return new PointF((float)Largeur, (float)Hauteur);
            }
        }
        /// <summary> chaine représentant le PointD </summary>
        public override string ToString()
        {
            return $"Width = {Largeur} Height = {Hauteur}";
        }
    }
}

File: PARTAGER/Structures/SizeD.vb
Friend Structure SizeD
    Private ReadOnly FlagNotEmpty As Boolean
    Friend Shared Function Add(Taille1 As SizeD, Taille2 As SizeD) As SizeD
        Return New SizeD(Taille1.Largeur + Taille2.Largeur, Taille1.Hauteur + Taille2.Hauteur)
    End Function
    Friend Shared Function Scale(Taille As SizeD, Facteur As Double) As SizeD
        Return New SizeD(Taille.Largeur * Facteur, Taille.Hauteur * Facteur)
    End Function
    Friend Shared ReadOnly Empty As SizeD
    Friend Sub New(Point As PointD)
        _Largeur = Point.X
        _Hauteur = Point.Y
        FlagNotEmpty = True
    End Sub
    Friend Sub New(Size As Size)
        _Largeur = Size.Width
        _Hauteur = Size.Height
        FlagNotEmpty = True
    End Sub
    Friend Sub New(Largeur As Double, Hauteur As Double)
        _Largeur = Largeur
        _Hauteur = Hauteur
        FlagNotEmpty = True
    End Sub
    Friend ReadOnly Property IsEmpty() As Boolean
        Get
            Return Not FlagNotEmpty
        End Get
    End Property
    ''' <summary> renvoie la largeur du rectangle </summary>
    Friend ReadOnly Property Largeur As Double
    ''' <summary> renvoie la hauteur du rectangle </summary>
    Friend ReadOnly Property Hauteur As Double
    ''' <summary> renvoie une size à partir du sizeD avec un arrondi Sup</summary>
    Friend ReadOnly Property ToSize() As Size
        Get
            Return New Size(CInt(Math.Round(_Largeur)), CInt(Math.Round(_Hauteur)))
        End Get
    End Property
    Friend ReadOnly Property ToPoint() As Point
        Get
            Return New Point(ToSize)
        End Get
    End Property
    ''' <summary> renvoie sizeF à partir du sizeD </summary>
    Friend ReadOnly Property ToSizeF() As SizeF
        Get
            Return New SizeF(CSng(_Largeur), CSng(_Hauteur))
        End Get
    End Property
    Friend ReadOnly Property ToPoinF() As PointF
        Get
            Return New PointF(CSng(_Largeur), CSng(_Hauteur))
        End Get
    End Property
    ''' <summary> chaine représentant le PointD </summary> 
    Public Overrides Function ToString() As String
        Return $"Width = {_Largeur} Height = {_Hauteur}"
    End Function
End Structure
